### ISSUE TITLE
bug: input getting hidden behind the keyboard on iOS

### ISSUE THREAD

**Issue by gerciljunio** [ionic-team/ionic-framework#27472]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

when there are inputs towards the end of an ion-content, when focusing on the input to type something it is hidden behind the keyboard, all types of modes have been tested in resizeMode in the plugin @capacitor/keyboard and none of them solved the problem .

tests were also carried out with setScroll and setAccessoryBarVisible and even so, still no solution.

short video showing the problem:

[short video showing the problem](https://github.com/ionic-team/ionic-framework/assets/4561073/a1ed65b5-0d6f-4549-b9e7-281205e08fce)

### Expected Behavior

when there is a focus on a text input that is at the end of the ion-content the expected result is that this field (input, textarea, etc.) is above the keyboard so that the user can see the field and what is being typed on it, on android everything works perfectly!

### Steps to Reproduce

1. pull the reproduction repo
2. install deps
3. run:

```
ionic build
ionic capacitor add ios
ionic capacitor run ios -l --external --verbose
```

4. select device (iPhone 13 with iOS 15.5 or iPhone 14 with iOS 16.2)  on list for the simulator via Xcode (14.2)
5. click on the last search entry at the bottom of the page, as shown in the video above
__

### Code Reproduction URL

https://github.com/gerciljunio/ionic-issue-input-hidden

### Ionic Info

Ionic:

   - Ionic CLI       : 7.1.1 (/usr/local/lib/node_modules/@ionic/cli)
   - Ionic Framework : @ionic/vue 7.0.5

Capacitor:

   - Capacitor CLI      : 5.0.1
   - @capacitor/android : 5.0.1
   - @capacitor/core    : 5.0.1
   - @capacitor/ios     : 5.0.3

Utility:

   - cordova-res : not installed globally
   - native-run  : 1.7.2

System:

   - NodeJS : v16.20.0 (/usr/local/Cellar/node@16/16.20.0/bin/node)
   - npm    : 8.19.4
   - OS     : macOS Monterey

### Additional Information

Test with simulator in Xcode 14.2:
iPhone 13 with iOS 15.5
iPhone 14 with iOS 16.2

on Android everything works perfectly!

I already looked for other issues here regarding my problem, but none of the solutions mentioned solved for these versions that I'm using. **maybe I missed something but anyway**, that's it!

thanks guys!

**Comment by liamdebeasi** [comment]
Thanks for the report. We use a "Scroll assist" utility to ensure that `ion-input` and `ion-textarea` instances do not get hidden by the keyboard when used in scrollable content. `ion-searchbar`, another type of input, does not participate in scroll assist because it is not meant to be used in the middle of scrollable content. The intended usage of `ion-searchbar` is that it is fixed to the top of the view. 

Our docs show how to use searchbars in toolbars: https://ionicframework.com/docs/api/searchbar#searchbars-in-toolbars

However, you could also use the searchbar in the content using `position: sticky` if you wanted it to scroll with your content.

**Comment by gerciljunio** [comment]
Thanks for the answer and sorry for the lack of attention related to input search, I did the test with input text and textarea and it worked to a certain extent, when the input is very close to the footer, even clicking on it, it only appears when typing something, this is this the expected behavior?  

But anyway i solved it by creating a helper class with keyboardWillShow event!

Once again thank you!

**Comment by gerciljunio** [comment]
> > I did the test with input text and textarea and it worked to a certain extent, when the input is very close to the footer, even clicking on it, it only appears when typing something, this is this the expected behavior?
> 
> Do you have a video of this happening?

Yes. My application will have many forms and fields to fill in, so I'm doing tests with input very close to the footer, this will be common in the application I'm developing. follow the video:

https://github.com/ionic-team/ionic-framework/assets/4561073/6b870b1b-cb60-498a-85ae-2a10270d7924

thanks @liamdebeasi

**Comment by liamdebeasi** [comment]
Thanks! Is that sample available in https://github.com/gerciljunio/ionic-issue-input-hidden?

**Comment by rainerfritz** [comment]
I can second that. It happens when you have a second input and that gets in focus. It seems like to rerender. That also happens in a modal for example. The whole modal gets lost. In addition the keyboard does not show the autocorrect pane on ion-input as it is set to autoCorrect='on'.

```
        <IonModal ref={modal} trigger="open-modal" initialBreakpoint={1} breakpoints={[0, 1]}>
          <div className="block">
            Direct Message
          <IonItem>
          <IonInput ref={callsignInputRef} label='To Callsign' labelPlacement="stacked" type='text' maxlength={MAX_CHAR_CALLSIGN} value={toCallsign} ></IonInput>
          </IonItem>
          <IonItem>
            
            <IonIcon slot='end' icon={caretForwardCircle} onClick={() => sendMsg(true)} color='primary' size='large'></IonIcon>
            <IonInput class="custominput" ref={textInputRef} label='Type Message' labelPlacement="stacked" type='text' autoCorrect='on' maxlength={MAX_CHAR_TEXTINPUT} aria-autocomplete='both'></IonInput>
          </IonItem>
          </div>
        </IonModal>
```

**Comment by liamdebeasi** [comment]
Thanks! Does this reproduce on a physical iOS device, or is it only happening on a simulator?

**Comment by gerciljunio** [comment]
I don't have a physical iOS device so I can only test on a simulator!  If you find that the physical device does not have this problem, please let me know! thanks!

**Comment by liamdebeasi** [comment]
I wasn't able to reproduce the issue on a physical iOS device. I know that testing keyboard behavior on an iOS simulator is not going to be exactly the same as testing on a physical iOS device, so that could be part of the problem here. I'm going to close this since I can't reproduce the issue on a physical device. However, if you find that this issue reproduces on a physical iOS device, let me know and I will be happy to reopen the issue. Thanks!
